nx-X11/programs/Xserver/Xext/shm.c

```diff
@@ -167,7 +166,7 @@ static Bool CheckForShmSyscall()
     int shmid = -1;
 
     /* If no SHM support in the kernel, the bad syscall will generate SIGSYS */
-    oldHandler = signal(SIGSYS, SigSysHandler);
+    oldHandler = OsSignal(SIGSYS, SigSysHandler);
```

Comment:
Phew... this one got really huge... and is still missing a lot of commit references for quite some of the `itimer` changes.

Also, I noticed `SMART_SCHEDULE_DEFAULT_INTERVAL` and `SMART_SCHEDULE_MAX_SLICE` changed upstream. If we change the default values as well, this might influence performance. No idea if for the better or worse, though.

One commit that should have been mentioned:

```
commit e10ba9e4b52269b2ac75c4802dce4ca47d169657
Author: Keith Packard <keithp@keithp.com>
Date:   Wed Nov 11 22:02:01 2015 -0800

    Remove non-smart scheduler. Don't require setitimer.

    This allows the server to call GetTimeInMillis() after each request is
    processed to avoid needing setitimer. -dumbSched now turns off the
    setitimer.

    Reviewed-by: Adam Jackson <ajax@redhat.com>
    Signed-off-by: Keith Packard <keithp@keithp.com>
```

It also looks like this hunk is missing from `nx-X11/programs/Xserver/hw/nxagent/NXdispatch.c`:

```diff
@@ -415,7 +418,10 @@ Dispatch(void)
                    result = BadLength;
                else
                    result = (* client->requestVector[MAJOROP])(client);
-
+
+               if (!SmartScheduleSignalEnable)
+                   SmartScheduleTime = GetTimeInMillis();
+
                if (result != Success)
                {
                    if (client->noClientException != Success)
```

But wait, there's more!

In `nx-X11/programs/Xserver/hw/nxagent/Display.h`:

```diff
 #define nxagentDisableTimer() \
 \
-    if (SmartScheduleTimerStopped == 0) \
-    { \
-      SmartScheduleStopTimer(); \
-    } \
-\
-    SmartScheduleDisable = 1;
+    SmartScheduleStopTimer(); \
+    SmartScheduleSignalEnable = FALSE;

 /*
  * File descriptor currently used by
```

`SmartScheduleSignalEnable` is set unguarded. This will go boom if `HAVE_SETITIMER` is undefined, because in that case `SmartScheduleSignalEnable` is macro'd to `FALSE` and we're then effectively setting `FALSE = FALSE`.